src/cobra/core/solution.py

```diff
@@ -1,32 +1,42 @@
-# -*- coding: utf-8 -*-
-
 """Provide unified interfaces to optimization solutions."""
 
-from __future__ import absolute_import
-
 import logging
-from builtins import object, super
+from typing import TYPE_CHECKING, Iterable
 
-from numpy import empty, nan
+import numpy as np
+import pandas as pd
 from optlang.interface import OPTIMAL
-from pandas import DataFrame, Series, option_context
 
-from cobra.util.solver import check_solver_status
+from ..util.solver import check_solver_status
+
+
+if TYPE_CHECKING:
+    from cobra import Metabolite, Model, Reaction
 
 
 __all__ = ("Solution", "get_solution")
 
-LOGGER = logging.getLogger(__name__)
+logger = logging.getLogger(__name__)
 
 
-class Solution(object):
+class Solution:
     """
     A unified interface to a `cobra.Model` optimization solution.
 
-    Notes
-    -----
-    Solution is meant to be constructed by `get_solution` please look at that
-    function to fully understand the `Solution` class.
+    Parameters
```

Comment:
Parameters are for the `__init__` method. I'm not sure why you move them here?

Reply:
From [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html#documenting-classes), I understand that the parameters for `__init__` need to be documented at the class level.

Reply:
I see 👍🏼